doc/style-guide.md

```diff
@@ -525,6 +525,62 @@ word within a compound word is capitalized except for the first word.
 * Any exceptions to these conventions should be flagged on the GitHub
   `agda-stdlib` issue tracker in the usual way.
 
+#### Fixity
+
+All functions and operators that are not purely prefix (typically
+anything that has a `_` in its name) should have an explicit fixity
+declared for it. The guidelines for these are as follows:
+
+General operations and relations:
+
+* binary relations of all kinds are `infix 4`
+
+* unary prefix relations `infix 4 ε∣_`
+
+* unary postfix relations `infixr 8 _∣0`
+
+* multiplication-like: `infixl 7 _*_`
+
+* addition-like  `infixl 6 _+_`
+
+* negation-like  `infix  8 ¬_`
+
+* and-like  `infixr 7 _∧_`
+
+* or-like  `infixr 6 _∨_`
+
+* post-fix inverse  `infix  8 _⁻¹`
+
+* bind `infixl 1 _>>=_`
+
+* list concat-like `infixr 5 _∷_`
```

Comment:
NB `Data.List.Base._++_` follows this scheme, yet it is also a `Monoid`-like addition operator (L544 above suggests it should therefore be `infixl 6 _++_`)... how should we resolve such inconsistencies?

Reply:
So I think that 'addition-like' means much closer to 'arithmetic' than something more general, i.e. that addition-like things could happen inside concat-like things without too much need for parens.

Or it could be that this is all a hack that happens to work by serendipitous accident rather than by conscious design!